server/plugins/task-plugin-api/src/main/java/com/oceanbase/odc/plugin/task/api/partitionplan/util/TimeDataTypeUtil.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.plugin.task.api.partitionplan.util;
+
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.oceanbase.odc.plugin.task.api.partitionplan.datatype.TimeDataType;
+
+/**
+ * {@link TimeDataTypeUtil}
+ *
+ * @author yh263208
+ * @date 2024-01-26 10:16
+ * @since ODC_release_4.2.4
+ */
+public class TimeDataTypeUtil {
+
+    public static Date getNextDate(Date base, int interval, int intervalPrecision) {
+        Calendar calendar = Calendar.getInstance();
```

Comment:
java 8 time API will be easier, e.g. LocalDateTime.plus()

Reply:
okay, I will learn this new class. btw, I am not familiar with this now~~😂 , so can we follow this current impl now?